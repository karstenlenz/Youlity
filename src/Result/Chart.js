import React, { useEffect, useRef, useState } from 'react'
import Chartjs from 'chart.js'

export default function Chart({ config }) {
  const chartContainer = useRef(null)
  const [chartInstance, setChartInstance] = useState(null)

  useEffect(() => {
    if (chartContainer && chartContainer.current) {
      const newChartInstance = new Chartjs(chartContainer.current, config)
      setChartInstance(newChartInstance)
    }
  }, [chartContainer, config])

  return (
    <div>
      <canvas width="345px" height="345px" ref={chartContainer} />
    </div>
  )
}
