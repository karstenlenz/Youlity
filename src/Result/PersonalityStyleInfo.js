import React from 'react'
import { useParams } from 'react-router-dom'
import styled from 'styled-components/macro'
import BackButton from '../common/BackButton'
import BulletList from '../common/BulletList'
import HeadlineUnderline from '../common/HeadlineUnderline'
import SectionBG from '../common/SectionBG'
import TextLink from '../common/TextLink'
import { personalityStyleData } from '../data/personalityStyleData'

export default function PersonalityStyleInfo() {
  const { styleId } = useParams()
  const currentStyleData = personalityStyleData[styleId - 1]

  return (
    <>
      <HeadlineUnderline>
        <h1>{currentStyleData.commonName}</h1>
      </HeadlineUnderline>
      <SectionNoBG>
        <HeadlineNoMarginTop>Einleitung</HeadlineNoMarginTop>
        <p>{currentStyleData.description}</p>
      </SectionNoBG>
      <InfoBG>
        <h2>Vorteile</h2>
        <BulletList listType="bullet_circle">
          {currentStyleData.strengths.map((strength) => {
            return <li key={strength}>{strength}</li>
          })}
        </BulletList>
      </InfoBG>
      <SectionNoBG>
        <h2>Nachteile</h2>
        <BulletList listType="bullet_circle">
          {currentStyleData.weaknesses.map((weakness) => {
            return <li key={weakness}>{weakness}</li>
          })}
        </BulletList>
      </SectionNoBG>
      <InfoBG>
        <h2>Reduzieren Sie Nachteile, indem Sie ...</h2>
        <BulletList listType="bullet_circle">
          {currentStyleData.tips.map((tip) => {
            return <li key={tip}>{tip}</li>
          })}
        </BulletList>
      </InfoBG>
      <h2>Weitere Informationen</h2>
      <p>
        Hinweis: folgende Ressourcen thematisieren größtenteils
        „Persönlichkeitsstörungen“. Das heißt weder, dass Sie eine
        Persönlichkeitsstörung haben, noch, dass nur Menschen mit
        Persönlichkeitsstörungen von den Informationen profitieren.
        Persönlichkeitsstörungen sind Steigerungen von Persönlichkeitsaspekten,
        die in geringerer Ausprägung viele Vorteile bieten können. Behalten Sie
        das bei der Lektüre im Hinterkopf.
      </p>
      <BulletList listType="bullet_circle">
        {currentStyleData.furtherInfo.map((info, index) => {
          return (
            <li key={index}>
              <TextLink href={info.url}>{info.text}</TextLink>
            </li>
          )
        })}
      </BulletList>
      <HelpSection>
        <h2>Hier gibt es Hilfe</h2>
        <p>
          Falls Sie unter Ihrem Persönlichkeitsstil leiden, holen Sie sich
          Hilfe! <br />
          Hier sind erste Anlaufstellen:
        </p>
        <ul>
          <li>
            <TextLink href="https://www.wege-zur-psychotherapie.org/">
              Infoseite "Wege zur Psychotherapie"
            </TextLink>
          </li>
          <li>
            <TextLink href="https://www.therapie.de/psyche/info/">
              Psychotherapeuten-Suche
            </TextLink>
          </li>
          <li>
            <TextLink href="tel:08001110111">
              Kostenlose Telefon-Seelsorge 0800-1110111
            </TextLink>
            <BackButton />
          </li>
        </ul>
      </HelpSection>
      <BottomSpacer />
    </>
  )
}

const InfoBG = styled(SectionBG)`
  margin-bottom: 0px;
  padding-bottom: 1px;
`

const HeadlineNoMarginTop = styled.h2`
  margin-top: 0px;
`
const SectionNoBG = styled.section`
  margin-bottom: 40px;
`

const HelpSection = styled.section`
  background: var(--secondary-disabled);
  padding: 15px;
  border-radius: 5px;
`
const BottomSpacer = styled.div`
  margin-bottom: 80px;
`
