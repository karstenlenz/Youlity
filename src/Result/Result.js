import React, { useState } from 'react'
import styled from 'styled-components'
import { styleData } from '../data/styleData'

export default function Result({ resultData }) {
  const [isNoticeVisible, setIsNoticeVisible] = useState(true)

  return (
    <>
      {isNoticeVisible && (
        <Notice>
          <button onClick={() => setIsNoticeVisible(false)}>X</button>
          <h3>Hinweis</h3>
          <p>
            Unser Testergebnis stellt nur eine Tendenz dar und ersetzt keine
            psychologische Beratung. Wenn Sie sich unwohl fühlen, holen Sie sich
            professionelle Hilfe.
          </p>
          <a href="https://www.psychenet.de/de/hilfe-finden/schnelle-hilfe/krisenanlaufstellen.html">
            Ansprechpartner
          </a>
        </Notice>
      )}

      <h1>Ergebnis</h1>
      <p>Sie haben eben Tests für folgende Persönlichkeitsstile ausgefüllt</p>
      <ul>
        {resultData.map((result) => {
          return (
            <li key={result.id}>
              <h3> {styleData[result.id - 1]?.name}</h3>
              <p>
                Sie haben {result.yesCount} von{' '}
                {styleData[result.id - 1].questions.length} Fragen mit "Ja"
                beantwortet.
              </p>
              {result.yesCount > 5 ? (
                <>
                  <p>
                    Das deutet darauf hin, dass dieser Stil bei Ihnen
                    überdurchschnittlich ausgeprägt ist. <br />
                    Das heißt nicht, dass etwas mit Ihnen „nicht stimmt“, oder
                    eine Störung vorliegt! Jeder Mensch hat einen oder mehrere
                    dominante Persönlichkeitsstile. Erst wenn Sie unter Ihrer
                    Persönlichkeit leiden, besteht Handlungsbedarf. <br />
                    Wir haben Ihnen die wichtigsten Informationen und Tipps zu
                    diesem Stil zusammengestellt.
                  </p>
                  <a href="#">Zu den Informationen</a>
                </>
              ) : (
                <p>
                  Das deutet <strong>nicht</strong> darauf hin, dass dieser Stil
                  bei Ihnen überdurchschnittlich ausgeprägt ist.
                </p>
              )}
            </li>
          )
        })}
      </ul>
      <p>
        Tipp: Wir speichern Ihr Ergebnis nicht, aber Sie können sich diese Seite
        im Browser als Lesezeichen speichern, um diese Informationen erneut
        aufzurufen.
      </p>
    </>
  )
}

const Notice = styled.section`
  background: #ddd;
  border: 1px solid black;
  padding: 15px;
  position: relative;

  button {
    position: absolute;
    right: 10px;
  }
`
