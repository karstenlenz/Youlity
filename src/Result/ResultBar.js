import React from 'react'
import styled, { keyframes } from 'styled-components/macro'

export default function ResultBar({ percentage, index }) {
  return (
    <Outline>
      <ColorBar percentage={percentage} index={index}>
        <Text>{percentage}% Ãœbereinstimmung</Text>
      </ColorBar>
    </Outline>
  )
}

const Text = styled.p`
  margin: 0;
  padding: 8px 15px;
`

const Outline = styled.div`
  border: 1px solid var(--dark-grey);
  border-radius: 5px;
  width: 100%;
`

const expandWidth = (width) => keyframes`
from {
      width: 0;
    }
    to {
      width: ${width}%;
    }
`

const ColorBar = styled.div`
  animation: ${(props) => expandWidth(props.percentage)} 0.8s
    ${(props) => 0.4 + 0.2 * props.index}s forwards ease-in-out;
  white-space: nowrap;
  overflow: visible;
  border-radius: 5px;
  height: 40px;
  width: 0;
  background-color: ${(props) =>
    props.percentage > 50
      ? 'rgba(0,197,170,' + (props.percentage / 50 - 1) + ')'
      : 'rgba(104,59,137,' + props.percentage / 100 + ')'};
`
