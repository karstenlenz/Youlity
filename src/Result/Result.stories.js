import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import Result from './Result'

export default {
  title: 'Result',
  component: Result,
}

export const ResultComponent = () => (
  <Router>
    <Result questionnaireIds="192837465" results="73" />
  </Router>
)
