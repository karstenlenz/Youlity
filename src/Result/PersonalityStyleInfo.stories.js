import React from 'react'
import PersonalityStyleInfo from './PersonalityStyleInfo'
import { BrowserRouter as Router } from 'react-router-dom'

export default {
  title: 'personalityStyleInfo ',
  component: 'personalityStyleInfo',
}

export const personalityStyleInfoComponent = () => {
  return (
    <Router path="/style-info/1">
      <PersonalityStyleInfo />
    </Router>
  )
}
