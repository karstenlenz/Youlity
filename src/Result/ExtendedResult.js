import PropTypes from 'prop-types'
import React from 'react'
import { Link } from 'react-router-dom'
import styled from 'styled-components/macro'
import Button from '../common/Button'
import HeadlineUnderline from '../common/HeadlineUnderline'
import SectionBGWithButton from '../common/SectionBGWithButton'
import TextLink from '../common/TextLink'
import { capitalizeFirstLetter } from '../common/util'
import { personalityStyleData } from '../data/personalityStyleData'
import ResultBar from './ResultBar'
import useResults from './useResults'

ExtendedResult.propTypes = {
  questionnaireIds: PropTypes.string.isRequired,
  results: PropTypes.string.isRequired,
}

export default function ExtendedResult({ questionnaireIds, results }) {
  const resultData = useResults(questionnaireIds, results)
  const sortedResultData = resultData
    .slice()
    .sort((a, b) => b.yesCount - a.yesCount)
  return (
    <>
      <img src="/img/style_info.svg" alt="" />
      <HeadlineUnderline>
        <h1>Ergebnis</h1>
      </HeadlineUnderline>
      <IntroText>
        Sie haben bisher folgende Fragebögen ausgefüllt. Klicken Sie auf einen
        Namen, um zu den Detail-Informationen zu gelangen.
      </IntroText>
      <section>
        {sortedResultData.map((result, index) => (
          <Link to={'/style-info/' + result.id} key={result.id}>
            <ResultSection>
              <ResultHeadline>
                {capitalizeFirstLetter(
                  personalityStyleData[result.id - 1]?.adjective
                )}{' '}
                {'>'}
              </ResultHeadline>
              <ResultBar
                percentage={(result.yesCount / 8) * 100}
                index={index}
              />
            </ResultSection>
          </Link>
        ))}
      </section>

      {9 - results.length > 0 ? (
        <>
          {9 - results.length === 1 ? (
            <p>Es gibt noch 1 weiteren Fragebogen.</p>
          ) : (
            <p>Es gibt noch {9 - results.length} weitere Fragebögen.</p>
          )}
          <Link to={'/questionnaire/entry/' + questionnaireIds + '/' + results}>
            <Button>Weiteren Fragebogen starten</Button>
          </Link>
          <SectionBGWithButton>
            <h2>Tagebuch</h2>
            <p>
              Sind Sie manchmal unzufrieden mit Ihrer Persönlichkeit? Unsere
              Tagebuch-Funktion hilft Ihnen, Ihr Verhalten zu analysieren und
              dadurch erste Schritte zu einer Veränderung anzustoßen.
            </p>
            <Link to="/journal">
              <Button btnType="secondary">Zum Tagebuch</Button>
            </Link>
          </SectionBGWithButton>
        </>
      ) : (
        <>
          <p>
            Sie haben die Fragebögen zu allen 9 Persönlichkeitsstilen
            ausgefüllt. Wir hoffen, dass Sie etwas Nützliches über Ihre
            Persönlichkeit mitnehmen konnten.
            <br />
            Wenn Sie noch mehr über sich erfahren wollen, nehmen Sie am besten
            Kontakt zu einem Psychotherapeuten auf.
          </p>
          <TextLinkWithMargin href="https://www.wege-zur-psychotherapie.org/">
            Infoseite "Wege zur Psychotherapie" &gt;
          </TextLinkWithMargin>
          <SectionBGWithButton>
            <h2>Tagebuch</h2>
            <p>
              Sind Sie manchmal unzufrieden mit Ihrer Persönlichkeit? Unsere
              Tagebuch-Funktion hilft Ihnen, Ihr Verhalten zu analysieren und
              dadurch erste Schritte zu einer Veränderung anzustoßen.
            </p>
            <Link to="/journal">
              <Button btnType="secondary">Zum Tagebuch</Button>
            </Link>
          </SectionBGWithButton>
          <BottomSpacer />
        </>
      )}
    </>
  )
}

const IntroText = styled.p`
  margin-bottom: 15px;
`
const TextLinkWithMargin = styled(TextLink)`
  margin: 15px 0;
`

const BottomSpacer = styled.div`
  margin-bottom: 40px;
`

const ResultSection = styled.section`
  margin-bottom: 20px;
`

const ResultHeadline = styled.h3`
  margin: 0 0 5px 0;
  text-align: left;
`
