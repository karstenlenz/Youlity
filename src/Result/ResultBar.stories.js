import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import ResultBar from './ResultBar'

export default {
  title: 'Result/ResultBar',
  component: ResultBar,
  argTypes: {
    percentage: {
      control: {
        type: 'number',
      },
    },
    index: {
      control: {
        type: 'number',
      },
    },
  },
}

export const ResultBarComponent = (args) => (
  <Router>
    <ResultBar {...args} />
  </Router>
)
ResultBarComponent.args = {
  percentage: 50,
  index: 1,
}
