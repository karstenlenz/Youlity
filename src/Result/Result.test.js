import { render } from '@testing-library/react'
import React from 'react'
import Result from './Result'
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useParams: jest.fn().mockReturnValue({ resultCode: '1&8&5&3' }),
}))

describe('The Result component', () => {
  it('renders correctly', () => {
    const { container } = render(<Result />)
    expect(container).toMatchSnapshot()
  })
})
