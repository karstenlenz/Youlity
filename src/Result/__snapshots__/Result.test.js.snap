// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Result component renders correctly 1`] = `
<div>
  <section
    class="NoticeBox__SectionStyled-sc-87rj1i-0 kftwUj"
  >
    <button
      class="NoticeBox__CloseButton-sc-87rj1i-1 eNuUKw"
    >
      <img
        alt="close overlay"
        src="/img/close.svg"
      />
    </button>
    <h3>
      Hinweis
    </h3>
    <p>
      Unser Testergebnis stellt nur eine Tendenz dar und ersetzt keine psychologische Beratung. Wenn Sie sich unwohl fühlen, holen Sie sich professionelle Hilfe.
      <br />
      <br />
    </p>
    <a
      class="TextLink__LinkStyled-sc-1a9sgio-0 eXUZha"
      href="https://www.wege-zur-psychotherapie.org/"
    >
      Infoseite "Wege zur Psychotherapie" &gt;
    </a>
  </section>
  <svg
    class="Result__IntroImgStyled-sc-19vma1i-0 jElWtw"
    title=""
  >
    style_info.svg
  </svg>
  <section
    class="HeadlineUnderline__HeadlineContainer-sc-1reeb6v-0 iPfevw"
  >
    <h1>
      Ergebnis
    </h1>
    <svg
      class="HeadlineUnderline__UnderlineStyled-sc-1reeb6v-1 iXilNe"
    >
      underline.svg
    </svg>
  </section>
  <p
    class="Result__ResultIntro-sc-19vma1i-1 ijCUam"
  >
    Sie haben eben Tests für folgende Persönlichkeitsstile ausgefüllt:
  </p>
  <section
    class="SectionBG__Background-sc-164jup7-0 OlfId"
  >
    <h3>
      Der selbstbewusste Persönlichkeitsstil
    </h3>
    <div
      class="ResultBar__Outline-sc-479x3f-1 echTYD"
    >
      <div
        class="ResultBar__ColorBar-sc-479x3f-2 gUwDrH"
      >
        <p
          class="ResultBar__Text-sc-479x3f-0 icLhoH"
        >
          75
          % Übereinstimmung
        </p>
      </div>
    </div>
    <p
      class="Result__StyleText-sc-19vma1i-2 brizaG"
    >
      Sie haben eine hohe Übereinstimmung mit diesem Persönlichkeitsstil. 
      <br />
      Das heißt nicht, dass etwas mit Ihnen „nicht stimmt“, oder eine Störung vorliegt! Jeder Mensch hat einen oder mehrere dominante Persönlichkeitsstile. Erst wenn Sie unter Ihrer Persönlichkeit leiden, besteht Handlungsbedarf. 
      <br />
      Lesen Sie sich die folgenden Informationen durch und prüfen Sie ehrlich, ob die Beschreibungen auf Sie zutreffen.
    </p>
    <a
      class="TextLink__LinkStyled-sc-1a9sgio-0 eXUZha"
      href="/style-info/1"
    >
      Zu den Informationen &gt;
    </a>
  </section>
  <section
    class="SectionBG__Background-sc-164jup7-0 OlfId"
  >
    <h3>
      Der sprunghafte Persönlichkeitsstil
    </h3>
    <div
      class="ResultBar__Outline-sc-479x3f-1 echTYD"
    >
      <div
        class="ResultBar__ColorBar-sc-479x3f-2 fqhHNn"
      >
        <p
          class="ResultBar__Text-sc-479x3f-0 icLhoH"
        >
          50
          % Übereinstimmung
        </p>
      </div>
    </div>
    <p
      class="Result__StyleText-sc-19vma1i-2 brizaG"
    >
      Das deutet 
      <strong>
        wahrscheinlich nicht
      </strong>
       darauf hin, dass dieser Stil bei Ihnen stark ausgeprägt ist.
      <br />
       Sie können sich die Beschreibung des Persönlichkeitsstil trotzdem gerne durchlesen und prüfen, ob Sie sich damit identifizieren können.
    </p>
    <a
      class="TextLink__LinkStyled-sc-1a9sgio-0 eXUZha"
      href="/style-info/9"
    >
      Zu den Informationen &gt;
    </a>
  </section>
  <h3>
    Weitere Fragebögen
  </h3>
  <p>
    Sie haben im ersten Schritt 2 von 9 Fragebögen ausgefüllt. Meist werden damit die für Sie relevantesten Persönlichkeitsstile abgedeckt. Sie können noch weitere Fragebögen ausfüllen, um zu erfahren, wie stark die weiteren Persönlichkeitsstile bei Ihnen ausgeprägt sind. Manche Menschen haben wenige, dafür stark ausgepräge Stile, andere mehrere, jeweils weniger stark ausgeprägte Stile.
  </p>
  <a
    href="/questionnaire/entry/1928373645/64"
  >
    <button
      class="Button__BtnStyled-sc-129yfz5-0 jyIaa-D"
      width="100"
    >
      Weiteren Fragebogen starten
    </button>
  </a>
  <section
    class="SectionBGWithButton__Background-sc-11viaht-0 bOMnEz"
  >
    <h2>
      Tagebuch
    </h2>
    <p>
      Sind Sie manchmal unzufrieden mit Ihrer Persönlichkeit? Unsere Tagebuch-Funktion hilft Ihnen, Ihr Verhalten zu analysieren und dadurch erste Schritte zu einer Veränderung anzustoßen.
    </p>
    <a
      href="/journal"
    >
      <button
        class="Button__BtnStyled-sc-129yfz5-0 cXymYf"
        width="100"
      >
        Zum Tagebuch
      </button>
    </a>
  </section>
  <p>
    Tipp: Wir speichern Ihr Ergebnis nicht, aber Sie können sich diese Seite im Browser als Lesezeichen speichern, um diese Informationen erneut aufzurufen.
  </p>
</div>
`;
